frame/contracts/src/wasm/mod.rs

```diff
@@ -1085,6 +1088,47 @@ mod tests {
 		assert_eq!(mock_ext.ecdsa_recover.into_inner(), [([1; 65], [1; 32])]);
 	}
 
+	#[test]
+	#[cfg(feature = "unstable-interface")]
+	fn contract_ecdsa_to_eth_address() {
+		/// calls `seal_ecdsa_to_eth_address` for the contstant and ensures the result equals the
+		/// expected one.
+		const CODE_ECDSA_TO_ETH_ADDRESS: &str = r#"
+(module
+	(import "__unstable__" "seal_ecdsa_to_eth_address" (func $seal_ecdsa_to_eth_address (param i32 i32)))
+	(import "env" "memory" (memory 1 1))
+
+	;; size of our buffer is 33 bytes
+	(data (i32.const 33) "\20")
```

Comment:
agreed on offset (should be `0`), but size imho shoud be `33 = 0x21` bytes (because it should take input public key of that size as an arg)

Reply:
This is the size of the **output** buffer. The input buffer size is implicit and not passed along. This still works because the buffer can be bigger than needed and also because the function (erroneously)  doesn't check the size of the output buffer.